src/parse_sync.cc

```diff
@@ -42,22 +40,30 @@ NAN_METHOD(protagonist::ParseSync) {
             Nan::ThrowTypeError(optionsResult->error);
         }
 
-        options = optionsResult->options;
-        astType = optionsResult->astType;
+        parseOptions = optionsResult->parseOptions;
+        serializeOptions = optionsResult->serializeOptions;
         FreeOptionsResult(&optionsResult);
     }
 
     // Parse the source data
-    snowcrash::ParseResult<snowcrash::Blueprint> parseResult;
-    snowcrash::parse(*sourceData, options | snowcrash::ExportSourcemapOption, parseResult);
-
-    Local<Object> result = v8_wrap(Result::WrapResult(parseResult, options, astType))->ToObject();
-
-    // Error Object (FIXME: Don't have this once we remove AST)
-    if (parseResult.report.error.code != snowcrash::Error::OK) {
-        Nan::ThrowError((Local<Value>) SourceAnnotation::WrapSourceAnnotation(parseResult.report.error));
-        return;
+    drafter_result *result;
+    int err_code = drafter_parse_blueprint(*sourceData,
+                                           &result,
+                                           parseOptions);
+    switch (err_code) {
+    case DRAFTER_EUNKNOWN:
+        Nan::ThrowError("Parser: Unknown Error");
+        break;
+    case DRAFTER_EINVALID_INPUT:
+        Nan::ThrowError("Parser: Invalid Input");
```

Comment:
Just have a proper sentence here instead of just saying "Parser: Invalid Input". For example, "The given input is not valid. Please check again."